### PR TITLE
Add evm builtins

### PR DESCRIPTION
**Pull request by chriseth**
Depends on #6774 

This is just the first step. Things to still take care of:

 - [x] optimizer should not use FunctionalInstruction anymore
 - [x] AsmParser should still parse non-functional instructions properly (or maybe retire?)

### CODE REVIEW

**Comment by chriseth** [comment]
This is still a little dirty at some points.

**Comment by chriseth** [comment]
Hm, I think it might be clean enough...

File: libyul/optimiser/Metrics.h [hunk 1]
```diff
@@ -71,16 +75,21 @@ class CodeSize: public ASTWalker
...
+	CodeCost(Dialect const& _dialect): m_dialect(_dialect) {}
+
 	void operator()(FunctionCall const& _funCall) override;
 	void operator()(FunctionalInstruction const& _instr) override;
 	void operator()(Literal const& _literal) override;
 	void visit(Statement const& _statement) override;
 	void visit(Expression const& _expression) override;
 
 private:
+	void instructionCost(dev::eth::Instruction _instruction);
```
Comment (ekpyron) [inline libyul/optimiser/Metrics.h]: A bit pedantic, but the name sounds like it returns the instruction cost... maybe ``addInstructionCost``? Or just ``operator()(dev::eth::Instruction)``?

**Review by ekpyron — DISMISSED** [review]
Looks good to me - only one pedantic minor comment.

**Pull request merged**